### ISSUE TITLE
HTTPClientExceptions code is 520

### ISSUE THREAD

**Issue by vlcinsky** [Pylons/pyramid#3212]
The docstring for [HTTPClientExceptions](https://github.com/Pylons/pyramid/blob/40d71e805bfcf8522c6af71995c05c496f1c4b4f/pyramid/httpexceptions.py#L607) states, 
> Unless specialized, this is a '400 Bad Request'

The truth is, this class is missing `code`, `title` and `explanation` so it inherits code 520 from parent error.

The same probably applies to `HTTPServerError`.

**Comment by tseaver** [comment]
@bertjwregeer You updated the base class's response code in 88fcdc4e.  Can you confirm that the docs should be adjusted accordingly?  Or should we override in the two subclasses?

**Comment by bertjwregeer** [comment]
It should be overridden in the two subclasses. The base class's response was modified because WebOb no longer allowed "None None" as a valid status code.

Basically what is in: https://github.com/Pylons/pyramid/blob/40d71e805bfcf8522c6af71995c05c496f1c4b4f/pyramid/httpexceptions.py#L611 should be hoisted up to the super class one further up, and things will be back to where they were.

Even before my change this would have been an issue, except that instead of sending back a `400 Bad Request` it would have sent back `None None`...

**Comment by mmerickel** [comment]
Maybe `HTTPBadRequest` should be an alias for `HTTPClientError`? It does seem bad that it's a 520 right now.

**Comment by mmerickel** [comment]
The only reason I'm against an alias is because I don't think it's smart to build an api where this works, even accidentally:

```python
foo = HTTPNotFound()
isinstance(foo, HTTPBadRequest) # -> True
```

With that in mind I'd just hoist the properties from HTTPBadRequest into HTTPClientError and let bad request inherit the defaults from it. So basically keep them separate classes but both looking like 4xx responses.

**Comment by stevepiercy** [comment]
Can this issue be closed, now that #3280, #3281, and #3282 have been merged?

**Comment by mmerickel** [comment]
Yep!
